src/components/Pen.js

```diff
@@ -4,19 +4,31 @@ import PropTypes from 'prop-types';
 import styles from './pen.cssmodule.styl';
 
 class Pen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      style: { backgroundColor: '' },
+      styleName: ''
+    };
+  }
   changeColor = () => {
     this.props.onChangeColor(this.props.color);
   }
+  updateStates(props = this.props) {
+    this.setState({
+      style: { backgroundColor: props.color },
+      styleName: 'pen' + (props.color === props.currentColor ? ' pen-active' : '')
+    });
+  }
+  componentWillMount() {
+    this.updateStates();
+  }
+  componentWillReceiveProps(nextProps) {
```

Comment:
I'm sorry, I forgot pushing.
Now, I've found better style to update computed states.
Please check better version.